File: app/auth/forgot-password/route.py
from fastapi import APIRouter, Depends, HTTPException
from pydantic import BaseModel
from sqlalchemy.orm import Session
import logging

from app import models
from app.database import get_db

router = APIRouter()
logger = logging.getLogger(__name__)

class ForgotPasswordRequest(BaseModel):
    email: str

@router.post("/forgot-password")
def forgot_password(
    request: ForgotPasswordRequest,
    db: Session = Depends(get_db)
):
    """
    Handle forgot password request
    In a real application, this would:
    1. Check if the email exists
    2. Generate a password reset token
    3. Send an email with the reset link
    """
    try:
        # Check if user exists
        user = db.query(models.User).filter(
            models.User.email == request.email
        ).first()
        
        if not user:
            # Don't reveal whether the email exists or not for security
            return {"message": "If the email exists, a reset link will be sent"}
        
        # In a real application, you would:
        # 1. Generate a unique token
        # 2. Store the token in the database with expiration
        # 3. Send an email with the reset link
        
        logger.info(f"Password reset requested for email: {request.email}")
        
        # For now, return success
        return {"message": "Password reset link has been sent to your email!"}
        
    except Exception as e:
        logger.error(f"Error in forgot password: {e}")
        raise HTTPException(status_code=500, detail="An error occurred")

File: frontend/app/api/auth/forgot-password/route.ts
import { NextResponse } from 'next/server';

export async function POST(request: Request) {
  try {
    const { email } = await request.json();

    if (!email) {
      return NextResponse.json(
        { error: 'Email is required' },
        { status: 400 }
      );
    }

    // In a real application, you would:
    // 1. Check if the email exists in your database
    // 2. Generate a password reset token
    // 3. Send an email with the reset link

    // For now, we'll simulate a successful response
    // The actual password reset would require:
    // - Database integration
    // - Email service integration (e.g., SendGrid, AWS SES)
    // - Token generation and storage

    // Simulate checking if user exists
    const backendUrl = process.env.NEXT_PUBLIC_BACKEND_URL || 'http://127.0.0.1:8000';
    
    try {
      const response = await fetch(`${backendUrl}/api/auth/forgot-password`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({ email }),
      });

      if (response.ok) {
        return NextResponse.json({ 
          message: 'Password reset link sent successfully' 
        });
      } else {
        const errorData = await response.json();
        return NextResponse.json(
          { error: errorData.error || 'Failed to send reset link' },
          { status: response.status }
        );
      }
    } catch (backendError) {
      // If backend is not available, return success anyway (for demo purposes)
      // In production, you would handle this differently
      console.log('Backend not available, simulating success');
      return NextResponse.json({ 
        message: 'Password reset link has been sent to your email!' 
      });
    }
  } catch (error) {
    console.error('Forgot password error:', error);
    return NextResponse.json(
      { error: 'An error occurred. Please try again.' },
      { status: 500 }
    );
  }
}
